src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -180,6 +197,44 @@ private static void WriteFunctionsMetricEvent(string executionId, ulong executio
             {
                 MetricEventSource.Log.RaiseFunctionsMetricEvent(executionId, executionTimeSpan, executionCount, executionStage);
             }
+
+            private void RaiseMetricsPerFunctionEvent()
+            {
+                List<FunctionMetrics> metricsEventsList = GetMetricsQueueSnapshot();
+
+                var aggregatedEventsPerFunction = from item in metricsEventsList
+                                                  group item by new { item.FunctionName } into FunctionGroups
```

Comment:
Any reason to create an anonymous type instance here? We should be able to group by the item.FunctionName, correct?